sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/statsbeat/_manager.py

```diff
@@ -0,0 +1,236 @@
+
+logger = logging.getLogger(__name__)
+
+
+class StatsbeatConfig:
+    """Configuration class for Statsbeat metrics collection."""
+    
+    def __init__(self, 
+                 endpoint: str,
+                 instrumentation_key: str,
+                 disable_offline_storage: bool = False,
+                 credential: Optional[Any] = None,
+                 distro_version: Optional[str] = None):
+        self.endpoint = endpoint
+        self.instrumentation_key = instrumentation_key
+        self.connection_string = _get_stats_connection_string(endpoint)
+        self.distro_version = distro_version
+        # features
+        self.disable_offline_storage = disable_offline_storage
+        self.credential = credential
+
+    @classmethod
+    def from_exporter(cls, exporter) -> 'StatsbeatConfig':
+        """Create configuration from an exporter instance."""
+        return cls(
+            endpoint=exporter._endpoint,
+            instrumentation_key=exporter._instrumentation_key,
+            disable_offline_storage=exporter._disable_offline_storage,
+            credential=exporter._credential,
+            distro_version=exporter._distro_version,
+        )
+    
+    def __eq__(self, other) -> bool:
+        """Compare two configurations for equality based on what can be changed via control plane."""
+        if not isinstance(other, StatsbeatConfig):
+            return False
+        return (
+            self.connection_string == other.connection_string and
+            self.disable_offline_storage == other.disable_offline_storage
+        )
+    
+    def __hash__(self) -> int:
+        """Hash based on connection string and offline storage setting."""
+        return hash((self.connection_string, self.disable_offline_storage))
+
+
+class StatsbeatManager(metaclass=Singleton):
+    """Thread-safe singleton manager for Statsbeat metrics collection with dynamic reconfiguration support."""
+
+    def __init__(self):
+        """Initialize instance attributes. Called only once due to Singleton metaclass."""
+        # Instance-level attributes
+        self._lock = threading.Lock()
+        self._initialized: bool = False  # type: ignore
+        self._metrics: Optional[_StatsbeatMetrics] = None  # type: ignore
+        self._meter_provider: Optional[MeterProvider] = None  # type: ignore
+        self._config: Optional[StatsbeatConfig] = None  # type: ignore
+    
+    def initialize(self, config: StatsbeatConfig) -> bool:
+        """Initialize statsbeat collection with thread safety."""
+        if not is_statsbeat_enabled():
+            return False
+            
+        with self._lock:
+            if self._initialized:
+                # If already initialized with the same config, return True
+                if self._config and self._config == config:
+                    return True
+                # If config is different, reconfigure
+                return self._reconfigure(config)
```

Comment:
This is not used currently but will be relevant once we have OneSettings change the config. Then we will be able to reconfigure and reinitialize stats during runtime if config changes.